src/track/beatgrid.cpp

```diff
@@ -40,75 +40,80 @@ class BeatGridIterator : public BeatIterator {
 };
 
 BeatGrid::BeatGrid(
-        const Track& track,
-        SINT iSampleRate)
-        : m_mutex(QMutex::Recursive),
-          m_iSampleRate(iSampleRate > 0 ? iSampleRate : track.getSampleRate()),
-          m_dBeatLength(0.0) {
+        SINT iSampleRate,
```

Comment:
Same here, we should ideally use the samplerate type unless it makes this PR much larger.

Reply:
I have intentionaly excluded unrelated refactoring. Because it will of cause makes this PR bigger.

I don't mind to add it here or in a follow up. What do you prefer?

Reply:
Then don't do it here. I just wanted to point it out because we should switch eventually.